dask/blockwise.py

```diff
@@ -659,9 +659,12 @@ def _cull_dependencies(self, all_hlg_keys, output_blocks):
         # Gather constant dependencies (for all output keys)
         const_deps = set()
         for (arg, ind) in self.indices:
-            if ind is None and isinstance(arg, str):
-                if arg in all_hlg_keys:
-                    const_deps.add(arg)
+            if ind is None:
+                try:
```

Comment:
Looks good to me.  However, could you explain how this change is captured by the modified `test_map_partitions_df_input`? (sorry if this is a silly question).

Reply:
It is not a silly question at all - by trial and error, this is the simplest that triggers the bug :)
The test is a heavily simplified implementation of https://github.com/rapidsai/gpu-bdb/blob/main/gpu_bdb/queries/q03/gpu_bdb_query_03.py.